extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -317,7 +379,9 @@ class AmpBrightcove extends AMP.BaseElement {
       videoId !== undefined
     ) {
       if (this.iframe_) {
-        this.iframe_.src = this.getIframeSrc_();
+        this.getIframeSrc_().then((src) => {
+          this.iframe_.src = src;
+        });
```

Comment:
This won't work--we can't have `mutatedAttributeCallback` be async. 

I'm thinking we can do something like this:
```
this.consentState = null;
this.sharedData = null;
this.consentString = null;

// Wait for consents and save them
getConsents() {
 ...
 Promise.all(consentsArray).then(arr => {
   this.consentState = arr[0];
   this.sharedData = arr[1];
   this.consentString = arr[2];
 });
}

// Should only be called after our consents have been stored
getIframeSrc_() {
 ...
 if (this.consentState) {
        src = addParamToUrl(
          src,
          'ampInitialConsentState',
          this.consentState
        );
      }
 ...
}

// Get consents then build the src
layoutCallback() {
   return this.getConsents().then(() => {
       const iframe = createFrameFor(this, this.getIframeSrc_());
       return loadPromise....
   }
}

// Change the src only once the consents have been save and the iframe has been created
mutatedAttributesCallback(mutations) {
 ...
 if (this.iframe_) {
  this.iframe_.src = this.getIframeSrc_();
 }
}
```

The idea here is to make getIframeSrc_ synchronous. This should be safe since, we only change the src within `mutatedAttributesCallback` once the iframe_ has been created in `layoutCallback`, at which point our consents will be stored.

Reply:
ok, refactored to work this way